src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -83,97 +80,93 @@ protected CodeGenerator(InsertionPoint insertionPoint, SelectionResult selection
-                    destination = previousMemberNode.Parent ?? previousMemberNode;
+                    var destination = mappedMember.Parent ?? mappedMember;
 
                     var info = codeGenerationService.GetInfo(
                         new CodeGenerationContext(
-                            afterThisLocation: previousMemberNode.GetLocation(),
+                            afterThisLocation: mappedMember.GetLocation(),
                             generateDefaultAccessibility: true,
                             generateMethodBodies: true),
                         Options,
-                        callSiteDocument.Project.ParseOptions);
-
-                    newContainer = codeGenerationService.AddMethod(destination, result.Data, info, cancellationToken);
-                }
+                        documentWithUpdatedCallSite.Project.ParseOptions);
 
-                var newSyntaxRoot = newCallSiteRoot.ReplaceNode(destination, newContainer);
-                var newDocument = callSiteDocument.Document.WithSyntaxRoot(newSyntaxRoot);
-
-                var generatedDocument = await SemanticDocument.CreateAsync(newDocument, cancellationToken).ConfigureAwait(false);
-
-                // For nullable reference types, we can provide a better experience by reducing use 
-                // of nullable reference types after a method is done being generated. If we can
-                // determine that the method never returns null, for example, then we can
-                // make the signature into a non-null reference type even though
-                // the original type was nullable. This allows our code generation to
-                // follow our recommendation of only using nullable when necessary.
-                // This is done after method generation instead of at analyzer time because it's purely
-                // based on the resulting code, which the generator can modify as needed. If return statements
-                // are added, the flow analysis could change to indicate something different. It's cleaner to rely
-                // on flow analysis of the final resulting code than to try and predict from the analyzer what 
-                // will happen in the generator. 
-                var finalDocument = await UpdateMethodAfterGenerationAsync(generatedDocument, result, cancellationToken).ConfigureAwait(false);
-                var finalRoot = finalDocument.Root;
-
-                var methodDefinition = finalRoot.GetAnnotatedNodesAndTokens(MethodDefinitionAnnotation).FirstOrDefault();
-                if (!methodDefinition.IsNode || methodDefinition.AsNode() == null)
-                {
-                    return await CreateGeneratedCodeAsync(
-                        result.Status.With(OperationStatus.FailedWithUnknownReason), finalDocument, cancellationToken).ConfigureAwait(false);
```

Comment:
this logic was entirely removed.  1. it shoudl not happen (and reflects us not having done our up front checks properly).  2. i made us resilient to this somehow happening, so that the extract method still succeeds, we just don't do things like offer the rename feature on it after extraction.